fictadvisor-api/src/v2/api/controllers/ScheduleController.ts

```diff
@@ -217,6 +217,51 @@ export class ScheduleController {
     return this.scheduleMapper.getEvent(result.event, result.discipline);
   }
 
+  @ApiBearerAuth()
+  @ApiOkResponse({
+    type: EventInfoResponse,
+  })
+  @ApiBadRequestResponse({
+    description: `\n
+    InvalidEntityIdException: 
+      Event with such id is not found
+      
+    InvalidWeekException:
+      Week parameter is invalid`,
+  })
+  @ApiUnauthorizedResponse({
+    description: `\n
+    UnauthorizedException:
+      Unauthorized`,
+  })
+  @ApiForbiddenResponse({
+    description: `\n
+    NoPermissionException:
+      You do not have permission to perform this action`,
+  })
+  @ApiParam({
+    name: 'eventId',
+    description: 'Id of the certain event',
+    required: true,
+  })
+  @ApiQuery({
+    name: 'week',
+    description: 'Week of the event',
+    required: true,
+  })
+  @ApiEndpoint({
+    summary: 'Get the information about event and all its pairs',
+    permissions: PERMISSION.GROUPS_$GROUPID_EVENT_INFOS_UPDATE,
+  })
+  @Get('/groups/:groupId/events/:eventId')
+  async getEventInfo (
+    @Param('eventId', EventByIdPipe) id: string,
+    @Query('week') week: number,
+  ) {
+    const result = await this.scheduleService.getEvent(id, week);
```

Comment:
`getEvent()` method isn't suitable, because there is a search for eventInfo by week index and it sets as 0 element of eventInfo array. And if it isn't there, it sets as null. But you need **all** eventInfos. So, I recommend you split that method into two.